backend/app/core/async_utils.py

```
@@ -0,0 +1,383 @@
+                e,
+                exc_info=True,
+            )
+        finally:
+            if callback:
+                try:
+                    callback(result, error)
+                except Exception as cb_error:
+                    logger.exception(
+                        "[async_utils] Callback for %s failed: %s",
+                        func.__name__,
+                        cb_error,
```

Ruff (0.14.14)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)